app/src/main/java/org/openobservatory/ooniprobe/client/callback/GetMeasurementCallback.java

```diff
@@ -0,0 +1,33 @@
+package org.openobservatory.ooniprobe.client.callback;
+
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonParser;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.io.IOException;
+
+import okhttp3.Call;
+import okhttp3.Callback;
+import okhttp3.Response;
+
+public abstract class GetMeasurementCallback implements Callback {
+    @Override
+    public void onResponse(@NotNull Call call, @NotNull Response response) {
+        try {
+            onSuccess(new GsonBuilder().setPrettyPrinting().disableHtmlEscaping().create()
+                    .toJson(new JsonParser().parse(response.body().string())));
+        } catch (Exception e) {
+            onError(e.getMessage());
+        }
```

Comment:
This pattern is dangerous. I will explain soon.

Reply:
```Java
    @Override
    public void onResponse(@NotNull Call call, @NotNull Response response) {
        String s;
        try {
            s = new GsonBuilder().setPrettyPrinting().disableHtmlEscaping().create()
                    .toJson(new JsonParser().parse(response.body().string()));
        } catch (Exception e) {
            onError(e.getMessage());
            return;
        }
        onSuccess(s);
```

Reply:
Basically, decouple callbacks and exceptions, because in general you want to call either onSuccess or onError and not both.